code/modules/mob/living/simple_animal/hostile/ordeal/violet.dm

```diff
@@ -17,40 +17,82 @@
 	guaranteed_butcher_results = list(/obj/item/food/meat/slab/fruit = 1)
 	damage_coeff = list(BRUTE = 1, RED_DAMAGE = 1, WHITE_DAMAGE = 1.5, BLACK_DAMAGE = 1, PALE_DAMAGE = 1)
 	blood_volume = BLOOD_VOLUME_NORMAL
+	var/list/enemies = list() //copying retaliate code cause i dont know how else to inherit it
 
 /mob/living/simple_animal/hostile/ordeal/violet_fruit/Initialize()
 	..()
-	addtimer(CALLBACK(src, .proc/ReleaseDeathGas), rand(60 SECONDS, 80 SECONDS))
+	addtimer(CALLBACK(src, .proc/ReleaseDeathGas), rand(60 SECONDS, 65 SECONDS))
+
+/mob/living/simple_animal/hostile/ordeal/violet_fruit/Found(atom/A)
+	if(isliving(A))
+		var/mob/living/L = A
+		if(!L.stat)
+			return L
+		else
+			enemies -= L
+	else if(ismecha(A))
+		var/obj/vehicle/sealed/mecha/M = A
+		if(LAZYLEN(M.occupants))
+			return A
+
+/mob/living/simple_animal/hostile/ordeal/violet_fruit/ListTargets()
+	if(!enemies.len)
+		return list()
+	var/list/see = ..()
+	see &= enemies // Remove all entries that aren't in enemies
+	return see
+
+/mob/living/simple_animal/hostile/ordeal/violet_fruit/adjustHealth(amount, updating_health = TRUE, forced = FALSE)
+	. = ..()
+	if(. > 0 && stat == CONSCIOUS)
+		Retaliate()
 
-/mob/living/simple_animal/hostile/ordeal/violet_fruit/CanAttack(atom/the_target)
-	return FALSE
+/mob/living/simple_animal/hostile/ordeal/violet_fruit/proc/Retaliate()
+	var/list/around = view(src, vision_range)
+
+	for(var/atom/movable/A in around)
+		if(A == src)
+			continue
+		if(isliving(A))
+			var/mob/living/M = A
+			if(faction_check_mob(M) && attack_same || !faction_check_mob(M))
+				enemies |= M
+		else if(ismecha(A))
+			var/obj/vehicle/sealed/mecha/M = A
+			if(LAZYLEN(M.occupants))
+				enemies |= M
+				enemies |= M.occupants
+
+/mob/living/simple_animal/hostile/ordeal/violet_fruit/AttackingTarget()
+	return
 
 /mob/living/simple_animal/hostile/ordeal/violet_fruit/Life()
 	. = ..()
 	if(!.) // Dead
 		return FALSE
-	new /obj/effect/temp_visual/revenant/cracks(get_turf(src))
-	for(var/mob/living/carbon/human/H in view(7, src))
-		new /obj/effect/temp_visual/revenant(get_turf(H))
-		H.apply_damage(6, WHITE_DAMAGE, null, H.run_armor_check(null, WHITE_DAMAGE))
+	new /obj/effect/temp_visual/small_smoke/second/fruit(get_turf(src))
+	for(var/mob/living/L in view(1, src))
```

Comment:
At the very least I feel it should be a size of 2 or 3. 1 means they're an issue one at a time and that kinda makes them push overs.